crates/core/c8y_smartrest/src/smartrest_deserializer.rs

```diff
@@ -304,7 +304,7 @@ impl SmartRestJwtResponse {
 /// let path_bufdate_time = get_datetime_from_file_path(&path).unwrap();
 /// ```
 pub fn get_datetime_from_file_path(
-    log_path: &PathBuf,
+    log_path: &Path,
```

Comment:
I wonder if there would be a benefit to make the signature more generic:

```suggestion
    log_path: impl AsRef<Path>,
```

Reply:
I am not sure, but I don't think so as all uses of the function are with path objects anyways... a `impl AsRef<Path>` would be beneficial if this function was called with `&str` or `String`, but it is not... :thinking:

Reply:
>  but it is not
It might be in the future.

Personally I would look at the rest of the crate and use whatever is more prevalent.

Reply:
I am happy to change it, if you want me to! :smile:

Reply:
That's ok, if not useful atm just merge it as is.